dali/pipeline/data/buffer.h

```diff
@@ -203,107 +211,81 @@ class Buffer {
   inline void set_type(const TypeInfo &new_type) {
     DALI_ENFORCE(IsValidType(new_type), "new_type must be valid type.");
     if (new_type == type_) return;
-
```

Comment:
These checks are meaningless now, that we can call `reserve` on a  `Buffer` without a valid type.